src/lua/init.lua

```diff
@@ -57,27 +57,29 @@ local function get_require_overload_count(stack)
     return 0
 end
 
--- Make path relative to the current directory.
-local function remove_root_directory(path)
-    local pwd = os.getenv("PWD")
-    if pwd == nil then
-        return path
+-- Return longest common subpath of two paths.
+local function get_common_subpath(path1, path2)
+    local result = '/'
+    path1 = path1:split('/')
+    path2 = path2:split('/')
+    for i = 1, math.min(#path1, #path2) do
+        if path1[i] == path2[i] then
+            result = fio.pathjoin(result, path1[i])
+        else
+            return result
+        end
     end
-    local cur_dir = pwd .. '/'
-    local start_len = #path
+    return result
+end
 
-    while #cur_dir > 0 and start_len == #path do
-        if cur_dir == '/' then
-            if path:sub(1, 1) == '/' then
-                path = path:sub(2)
-            end
-            break
-        end
-        path = path:gsub(cur_dir, '')
-        local ind = cur_dir:find('[^/]+/$')
-        cur_dir = cur_dir:sub(1, ind - 1)
+-- Strip a common part of cwd and path from the path.
+local function strip_cwd_from_path(cwd, path)
+    if not cwd or cwd:sub(1, 1) ~= '/' or
+       not path or path:sub(1, 1) ~= '/' then
+        return path
     end
-    return path
+    local common = get_common_subpath(cwd, path)
+    return path:gsub('^' .. common .. '/?', '')
```

Comment:
It doesn't look like you may use regular expression like that: if I update the test as follows it starts to fail although it shouldn't.

<details><summary>diff</summary>

```diff
diff --git a/test/app-luatest/per_module_log_unit_test.lua b/test/app-luatest/per_module_log_unit_test.lua
index ee6b1b85b598..81f45bb85d71 100644
--- a/test/app-luatest/per_module_log_unit_test.lua
+++ b/test/app-luatest/per_module_log_unit_test.lua
@@ -13,7 +13,7 @@ g.test_strip_cwd_from_path = function()
         { cwd = '/totally/different', path = '/home/aaa/bbb', expected = 'home/aaa/bbb' },
         { cwd = '/doesnt/matter', path = 'aaa/bbb', expected = 'aaa/bbb' },
         { cwd = '/qwerty', path = '/aaa/qwerty/bbb', expected = 'aaa/qwerty/bbb' },
-        { cwd = '/home/aaa/bbb', path = '/home/aaa/bbbccc/ddd', expected = 'bbbccc/ddd' },
+        { cwd = '/home/a-aa/bbb', path = '/home/a-aa/bbbccc/ddd', expected = 'bbbccc/ddd' },
         { cwd = '/home/common_qwe', path = '/home/common_asd/aaa', expected = 'common_asd/aaa' },
     }
     for _, test in pairs(testcases) do
```

</details>

Reply:
Thanks, fixed.